fasm2bels/models/dsp_models.py

```diff
@@ -0,0 +1,614 @@
+                        if top.find_source_from_sink(
+                                site, 'CARRYINSEL[2]'
+                        ) == 0 and top.find_source_from_sink(
+                                site, 'CARRYINSEL[1]'
+                        ) == 0 and top.find_source_from_sink(
+                                site, 'CARRYINSEL[0]') == 1:
+                            dsp_in_use = True
+
+                        if top.find_source_from_sink(
+                                site, 'CARRYINSEL[2]'
+                        ) == 0 and top.find_source_from_sink(
+                                site, 'CARRYINSEL[1]'
+                        ) == 1 and top.find_source_from_sink(
+                                site, 'CARRYINSEL[0]') == 1:
+                            dsp_in_use = True
+
+                        if top.find_source_from_sink(
+                                site, 'CARRYINSEL[2]'
+                        ) == 0 and top.find_source_from_sink(
+                                site, 'CARRYINSEL[1]'
+                        ) == 1 and top.find_source_from_sink(
+                                site, 'CARRYINSEL[0]') == 0:
+                            dsp_in_use = True
+
+                        # Refer table 2-9 in ug479_7Series_DSP48E1 manual
+                        if (top.find_source_from_sink(site, 'OPMODE[6]') ^ int(
+                                dsp_bel.parameters['IS_OPMODE_INVERTED'][-1])
+                            ) == 1 and (
+                                top.find_source_from_sink(site, 'OPMODE[5]') ^
+                                int(dsp_bel.parameters['IS_OPMODE_INVERTED']
+                                    [-2])
+                            ) == 0 and (
+                                top.find_source_from_sink(site, 'OPMODE[4]') ^
+                                int(dsp_bel.parameters['IS_OPMODE_INVERTED']
+                                    [-3])
+                            ) == 0 and (
+                                top.find_source_from_sink(site, 'OPMODE[3]') ^
+                                int(dsp_bel.parameters['IS_OPMODE_INVERTED']
+                                    [-4])
+                            ) == 1 and (
+                                top.find_source_from_sink(site, 'OPMODE[2]') ^
+                                int(dsp_bel.parameters['IS_OPMODE_INVERTED']
+                                    [-5])
+                            ) == 0 and (
+                                top.find_source_from_sink(site, 'OPMODE[1]') ^
+                                int(dsp_bel.parameters['IS_OPMODE_INVERTED']
+                                    [-6])
+                            ) == 0 and (
+                                top.find_source_from_sink(site, 'OPMODE[0]') ^
+                                int(dsp_bel.parameters['IS_OPMODE_INVERTED']
+                                    [-7])) == 0:
+                            dsp_in_use = True
+
+                    break

```

Comment:
I believe that these two cases share a lot of common code. Perhaps the common checks can be isolated into a function?

I also find those big multi-line ifs awkward. You could implement that in a for loop even taking advantage of Python `for`/`else` construct

Reply:
Fixed it. Can you have a look?

Reply:
Alright, looks good.